src/Compilers/Core/MSBuildTask/CommandLineBuilderExtension.cs

```diff
@@ -263,5 +263,14 @@ internal void AppendSwitchIfNotNull
                 }
             }
         }
+
+        internal void AppendArgumentsIfNotNull(string arguments)
+        {
+            if (!string.IsNullOrEmpty(arguments))
+            {
+                AppendSpaceIfNotEmpty();
+                AppendTextUnquoted(arguments);
```

Comment:
>AppendTextUnquoted [](start = 16, length = 18)

It's odd that we expect the arguments to be already quoted. Wouldn't it make more sense to quote them as we are adding them to the command line?

Reply:
Updated to add as quoted text.